### PR TITLE
CCN NPI fixes

### PR DESCRIPTION
**Pull request by okeefm**
Built a better system for adding CCNs/NPIs to Cat I headers. Basically, the system looks for the given `root` ID in the Provider Performances for a patient. If that's not found, it puts in the default.

### CODE REVIEW

File: exporter/templates/cat1/r3/cat1.xml [hunk 1]
```diff
@@ -86,16 +86,9 @@
...
-        {{if .CMSCompatibility}}
-          {{range .Record.ProviderPerformances}}
-            {{range .Provider.CDAIdentifiers}}
-              {{if eq .Root "2.16.840.1.113883.4.6"}}
-                <id root="2.16.840.1.113883.4.6" extension="{{.Extension}}" />
-              {{end}}
-            {{end}}
-          {{end}}
+        {{range .Record.ProviderIdentifiersForRoot "2.16.840.1.113883.4.6" "1234567893" }}
+          <id root="2.16.840.1.113883.4.6" extension="{{.}}" />
```
Comment (tstrass) [inline exporter/templates/cat1/r3/cat1.xml:91]: Just going off of [this template in HDS](https://github.com/projectcypress/health-data-standards/blob/master/templates/cat1/r3/show.cat1.erb#L87), these provider ID's should only be there if `CMSCompatibility` is on (and same for R3.1, R4)

File: models/provider.go [hunk 2]
```diff
@@ -13,3 +13,25 @@ type Provider struct {
...
+	}
+	return ""
+}
+
+// IdentifiersForRoot applies IdentifierForRoot to a passed-in slice of Providers
+func IdentifiersForRoot(prov []Provider, root string) []string {
+	ids := []string{}
+	for _, provider := range prov {
+		str := provider.IdentifierForRoot(root)
+		if str != "" {
```
Comment (tstrass) [inline models/provider.go]: Minor nitpicks but can you make the assignment to str an initialization statement for the if? And maybe rename `str` to `rootID` or `id`, and `prov` to `provs`?

**Review by tstrass — CHANGES_REQUESTED** [review]

File: exporter/templates/cat1/r3/cat1.xml (hunk 1 above)
Comment (okeefm) [reply exporter/templates/cat1/r3/cat1.xml:91]: The provider IDs are allowed by the HL7 Implementation Guide, they are just only required by the CMS IG. Cypress generates NPIs for all providers, but if there isn't an NPI, this function will insert "1234567893" instead, which is a valid NPI.

**Review by tstrass — CHANGES_REQUESTED** [review]
Needs rebase+make and the issue Dave pointed out with calling `Record.ProviderIdentifiersForRoot` needs to be addressed

**Review by dczulada — APPROVED** [review]
Appears to work

**Review by tstrass — APPROVED** [review]

**Pull request merged**